test/CMakeLists.txt

```diff
@@ -75,6 +75,11 @@ foreach(name IN LISTS tests)
       FunctionalPlus::fplus
       doctest::doctest
   )
+
+  if(TARGET auto_generate)
+    add_dependencies("${name}" auto_generate)
```

Comment:
This should go in the main CML unconditionally after the custom target.

Reply:
> This should go in the main CML unconditionally after the custom target

We are inside a foreach loop inside tests/CMakeList that produces multiple targets which are unknown to the main CMakeLists. I do not see an obvious way for how this could be done in the main  CML

Reply:
`add_dependencies(fplus auto_generate)`

Reply:
Ok, thanks & done ;-)